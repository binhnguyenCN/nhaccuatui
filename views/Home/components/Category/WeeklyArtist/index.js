import React from 'react';

import categoryStyles from '../styles.module.scss';
import styles from './styles.module.scss';

const WeeklyArtist = () => (
  <article
    className={`${categoryStyles.category}
    ${styles['weekly-artist-category']}`}
  >
    <h1>WEEKLY ARTIST CATEGORY</h1>
  </article>
);
export default WeeklyArtist;
