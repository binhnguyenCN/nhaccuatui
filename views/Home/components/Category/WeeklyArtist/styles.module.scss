.weekly-artist-category {
  grid-area: weekly;
}
