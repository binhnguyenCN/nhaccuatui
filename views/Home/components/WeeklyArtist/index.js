import React from 'react';

import styles from './styles.module.scss';

const WeeklyArtist = () => (
  <article className={`${styles['weekly-artist-category']}`}>
    <h1>WEEKLY ARTIST CATEGORY</h1>
  </article>
);
export default WeeklyArtist;
