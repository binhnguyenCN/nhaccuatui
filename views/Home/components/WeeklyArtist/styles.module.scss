.weekly-artist-category {
  grid-area: weekly;
  background-color: blue;
}
