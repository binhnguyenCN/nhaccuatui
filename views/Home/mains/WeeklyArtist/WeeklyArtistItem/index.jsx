// libs
import Link from 'next/link';
import React from 'react';
// others
import styles from './styles.module.scss';

const WeeklyArtistItem = ({ artist }) => (
  <li className={styles['weekly-artist-item']} title={artist.title}>
    <div className={styles['weekly-artist-thumbnail']}>
      <Link href={artist.thumbnail} passHref>
        <span>
          <img
            src={artist.thumbnail}
            width={152}
            height={152}
            alt={artist.title}
          />
        </span>
      </Link>
      <div className={styles['thumbnail-dark-layer']} />
    </div>
  </li>
);

export default WeeklyArtistItem;
