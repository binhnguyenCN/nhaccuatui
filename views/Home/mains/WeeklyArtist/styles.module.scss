.weekly-artist-wrapper {
  grid-area: weekly;
}
