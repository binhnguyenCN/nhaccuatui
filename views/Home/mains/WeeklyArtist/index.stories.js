import React from 'react';

import WeeklyArtist from '.';
import { topSong } from '../../../../api/mockApi';

export default {
  title: 'Pages/Home/main/WeeklyArtist',
  component: WeeklyArtist,
  parameters: { artists: topSong },
};
export const Template = () => <WeeklyArtist artists={topSong} />;
