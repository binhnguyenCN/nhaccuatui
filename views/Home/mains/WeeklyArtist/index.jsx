// libs
import React from 'react';
// layouts
import CategoryList from '../../components/CategoryList';
import CategoryTitle from '../../components/CategoryTitle';
import WeeklyArtistItem from './WeeklyArtistItem';
// others
import styles from './styles.module.scss';
import CategoryTitles from '@/data_source/CategoryTitle';

const WeeklyArtist = ({ artists }) => (
  <article className={styles['weekly-artist-wrapper']}>
    <div className={styles['weekly-artist-inner']}>
      <CategoryTitle title={CategoryTitles.weeklyArtist} />
      <CategoryList>
        {artists.map((artist, index) => {
          if (index < 5) {
            return (
              <WeeklyArtistItem
                title={artist.title}
                thumbnail={artist.thumbnail}
                key={artist.key}
              />
            );
          }
          return null;
        })}
      </CategoryList>
    </div>
  </article>
);
export default WeeklyArtist;
