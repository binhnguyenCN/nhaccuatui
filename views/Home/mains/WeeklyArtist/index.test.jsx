// libs
import renderer from 'react-test-renderer';
// layouts
import WeeklyArtist from '.';
// mocks
import mockData from '@/mocks/mockApi';

describe('render weekly artists', () => {
  it('should renders correctly', () => {
    // act
    const tree = renderer
      .create(<WeeklyArtist artists={mockData.topSong} />)
      .toJSON();
    // assert
    expect(tree).toMatchSnapshot();
  });
});
