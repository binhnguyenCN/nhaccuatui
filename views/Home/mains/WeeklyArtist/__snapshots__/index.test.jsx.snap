// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render weekly artists should renders correctly 1`] = `
<article
  className="weekly-artist-wrapper"
>
  <div
    className="weekly-artist-inner"
  >
    <h3
      className="category-title"
    >
      <span
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span>
          weekly artist chart
        </span>
        <span>
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </span>
    </h3>
    <ul
      className="category-list"
    >
      <li
        className="weekly-artist-item"
        title="Top 100 Rap Việt Hay Nhất"
      >
        <div
          className="weekly-artist-thumbnail"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2021/04/07/2/7/0/9/1617776161396_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Top 100 Rap Việt Hay Nhất"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2021/04/07/2/7/0/9/1617776161396_300.jpg"
              width={152}
            />
          </span>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
      </li>
      <li
        className="weekly-artist-item"
        title="Top 100 Rap/ Hip Hop Hay Nhất"
      >
        <div
          className="weekly-artist-thumbnail"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2021/05/26/b/e/2/9/1622014521880_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Top 100 Rap/ Hip Hop Hay Nhất"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2021/05/26/b/e/2/9/1622014521880_300.jpg"
              width={152}
            />
          </span>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
      </li>
      <li
        className="weekly-artist-item"
        title="Top 100 Nhạc Latin Hay Nhất"
      >
        <div
          className="weekly-artist-thumbnail"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2021/05/26/b/e/2/9/1622014540231_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Top 100 Nhạc Latin Hay Nhất"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2021/05/26/b/e/2/9/1622014540231_300.jpg"
              width={152}
            />
          </span>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
      </li>
      <li
        className="weekly-artist-item"
        title="Top 100 Electronica/Dance Hay Nhất"
      >
        <div
          className="weekly-artist-thumbnail"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2020/09/16/e/1/f/f/1600239245257_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Top 100 Electronica/Dance Hay Nhất"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2020/09/16/e/1/f/f/1600239245257_300.jpg"
              width={152}
            />
          </span>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
      </li>
      <li
        className="weekly-artist-item"
        title="Top 100 Blues/Jazz Hay Nhất"
      >
        <div
          className="weekly-artist-thumbnail"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2020/09/16/e/1/f/f/1600250971140_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Top 100 Blues/Jazz Hay Nhất"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2020/09/16/e/1/f/f/1600250971140_300.jpg"
              width={152}
            />
          </span>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
      </li>
    </ul>
  </div>
</article>
`;
